wordpress-mobile/WordPress-Android
# Issue/1852 gutenberg mobile retry media

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/1852

Note: ~Since this PR targets a freezed branch, it'd be good to have it reviewed by two developers.~ One reviewer is enough.

Note 2: I'd recommend setting the GitHub diff tool to "ignore whitespace changes" to make the review easier.

The issue was that [this line](https://github.com/wordpress-mobile/WordPress-Android/blob/5fa7d2310fb1de9caeec10267881d0a0d2650b2d/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostRepository.kt#L125) was invoked only when there was a difference between the PostModel and the UI content. When the user clicks on "Retry", the content on the UI doesn't change, therefore [this lambda](https://github.com/wordpress-mobile/WordPress-Android/blob/4cc6a359ba121818c44f72e01b485a1001a1f815/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java#L1534) was never invoked and as a result the UploadService was never started [here](https://github.com/wordpress-mobile/WordPress-Android/blob/65204adbee0da2e864d387a7579786ac46062c50/WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/UploadMediaUseCase.kt#L27).

The fix replaces [`onSuccess` with `onCompleted`](https://github.com/wordpress-mobile/WordPress-Android/blob/20a93a60f1c0a89226a9a4de5b299fe4931342c5/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostRepository.kt#L129) lambda, which is always invoked with either `UpdatePostResult.Updated` or `UpdatePostResult.NoChanges`.



To test:
1.  Start a new post and add an image block
2. Add image from "Choose from device" with airplane mode on (observe failure message)
3. Turn airplane mode off and tap image
4. Tap retry upload
5. Observe the image upload is started

PR submission checklist:


~cc @maxme I think it'd be good to merge this into 14.1 as not being able to retry media upload can be frustrating. (Even though the user can always leave the editor and let the app handle the retry.)~

Props to @mkevins for the great initial investigation on the ticket;)!
